doc/exts/pylint_messages.py

```diff
@@ -0,0 +1,227 @@
+
+import os
+from collections import defaultdict
+from typing import DefaultDict, Dict, List, NamedTuple, Optional, Tuple
+
+from sphinx.application import Sphinx
+
+from pylint.checkers import initialize as initialize_checkers
+from pylint.constants import MSG_TYPES
+from pylint.extensions import initialize as initialize_extensions
+from pylint.lint import PyLinter
+from pylint.message import MessageDefinition
+from pylint.utils import get_rst_title
+
+# PACKAGE/docs/exts/pylint_extensions.py --> PACKAGE/
+PYLINT_BASE_PATH = os.path.dirname(
+    os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+)
+"""Base path to the project folder"""
+
+
+MSG_TYPES_DOC = {k: v if v != "info" else "information" for k, v in MSG_TYPES.items()}
+
+
+class MessageData(NamedTuple):
+    checker: str
+    id: str
+    name: str
+    definition: MessageDefinition
+
+
+MessagesDict = Dict[str, List[MessageData]]
+OldMessagesDict = Dict[str, DefaultDict[Tuple[str, str], List[Tuple[str, str]]]]
+"""DefaultDict is indexed by tuples of (old name symbol, old name id) and values are
+tuples of (new name symbol, new name category)
+"""
+
+
+def _register_all_checkers_and_extensions(linter: PyLinter) -> None:
+    """Registers all checkers and extensions found in the default folders."""
+    initialize_checkers(linter)
+    initialize_extensions(linter)
+
+
+def _get_all_messages(
+    linter: PyLinter,
+) -> Tuple[MessagesDict, OldMessagesDict]:
+    """Get all messages registered to a linter and return a dictionary indexed by message
+    type.
+    Also return a dictionary of old message and the new messages they can be mapped to.
+    """
+    messages_dict: MessagesDict = {
+        "fatal": [],
+        "error": [],
+        "warning": [],
+        "convention": [],
+        "refactor": [],
+        "information": [],
+    }
+    old_messages: OldMessagesDict = {
+        "fatal": defaultdict(list),
+        "error": defaultdict(list),
+        "warning": defaultdict(list),
+        "convention": defaultdict(list),
+        "refactor": defaultdict(list),
+        "information": defaultdict(list),
+    }
+    for message in linter.msgs_store.messages:
+        message_data = MessageData(
+            message.checker_name, message.msgid, message.symbol, message
+        )
+        messages_dict[MSG_TYPES_DOC[message.msgid[0]]].append(message_data)
+
+        if message.old_names:
+            for old_name in message.old_names:
+                category = MSG_TYPES_DOC[old_name[0][0]]
+                old_messages[category][(old_name[1], old_name[0])].append(
+                    (message.symbol, MSG_TYPES_DOC[message.msgid[0]])
+                )
+
+    return messages_dict, old_messages
+
+
+def _write_message_page(messages_dict: MessagesDict) -> None:
+    """Create or overwrite the file for each message."""
+    for category, messages in messages_dict.items():
+        category_dir = os.path.join(PYLINT_BASE_PATH, "doc", "messages", category)
```

Comment:
```suggestion
        category_dir = DOC_MESSAGES_PATH / category
```

I would create a constant `DOC_MESSAGES_PATH = PYLINT_BASE_PATH / "doc" / "messages"`, this seems to be used multiple time.

Reply:
Done 👍